service/storage_proxy.cc

```diff
@@ -582,102 +582,110 @@ class storage_proxy::remote {
         });
     }
 
-    future<rpc::tuple<foreign_ptr<lw_shared_ptr<query::result>>, cache_temperature, replica::exception_variant>>
-    handle_read_data(
-            const rpc::client_info& cinfo, rpc::opt_time_point t,
-            query::read_command cmd1, ::compat::wrapping_partition_range pr,
-            rpc::optional<query::digest_algorithm> oda, rpc::optional<db::per_partition_rate_limit::info> rate_limit_info_opt) {
+    enum class read_verb {
+        read_data,
+        read_mutation_data,
+        read_digest
+    };
+    friend std::ostream& operator<<(std::ostream& os, const read_verb& verb) {
+        switch (verb) {
+            case read_verb::read_data:
+                os << "read_data";
+                break;
+            case read_verb::read_mutation_data:
+                os << "read_mutation_data";
+                break;
+            case read_verb::read_digest:
+                os << "read_digest";
+                break;
+        }
+        return os;
```

Comment:
New code should use fmt formatters. Don't make more work for @tchaikov .

Reply:
should this `::format("{}", read_verb::read_data)` just work or should I implement something?

Reply:
See 3928a9a4e9bb3e0f6caea09bba9faee275044d01 for an example (and https://fmt.dev/latest/api.html#formatting-user-defined-types for the theory)

Reply:
This `read_verb` is a local, ad-hoc, private enum, nested in some class, I can't partially specialise a class inside another class and I don't want to make this enum more public than it's needed.

@tchaikov What's the problem with `operator<<`? I would prefer to turn it to just local lambda `read_verb->string_view` if it's so bad.